### PR TITLE
Add more ruby text objects

### PR DESCRIPTION
**Pull request by jeetelongname**
I have added all the inner variants of the supplied text objects as well
as 
- adding modifier variants of conditionals and loops, 
- comments and
- parameters support

I have tried keep these query's efficient but there may be room for improvement
[](https://github.com/hlissner/doom-emacs/pull/5401)

### CODE REVIEW

File: queries/ruby/textobjects.scm [hunk 1]
```diff
@@ -1,28 +1,80 @@
...
+  method: (_) @call.inner ) @call.outer
+
+;; classes
 [
-  (class)
-  (module)
+  (class
+    name: (constant)
+    superclass: (_)
+    (_) @class.inner)
```
Comment (marocchino) [inline queries/ruby/textobjects.scm:14]: `class` could be has no parents.
```suggestion
  (class
    name: (constant)
    superclass: (_)
    (_) @class.inner)
  (class
    name: (constant)
    !superclass
    (_) @class.inner)
```

Comment (jeetelongname) [reply queries/ruby/textobjects.scm:14]: Thanks. this was a bit of oversight on my part

**Comment by jeetelongname** [comment]
Any update on this?

**Comment by jeetelongname** [comment]
Thanks!

**Pull request merged**